### ISSUE TITLE
WARNING: Illegal reflective access by com.google.inject.assistedinject.FactoryProvider2

### ISSUE THREAD

**Issue by natros** [mcculls/guice-betas#1]
Hi, 

I am still getting the following warning. Is that normal?

WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.assistedinject.FactoryProvider2 (file:/Users/fsousa/.gradle/caches/modules-2/files-2.1/com.google.inject.extensions/guice-assistedinject/4.2.4-20200419-NEWAOP-BETA/32713c29b98cc17976fef229ef62fa4813c61d82/guice-assistedinject-4.2.4-20200419-NEWAOP-BETA.jar) to constructor java.lang.invoke.MethodHandles$Lookup(java.lang.Class,int)
WARNING: Please consider reporting this to the maintainers of com.google.inject.assistedinject.FactoryProvider2
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

**Comment by mcculls** [comment]
Yes, that is still expected to happen for the AssistedInject extension because it has code that tries different APIs to see what's available for introspecting and calling default methods. It won't affect operation of AssistedInject because it always falls back to a sane approach.

Note this is totally separate to the AOP provided by Guice core, which is why the new AOP code in core doesn't change the above message from AssistedInject.

I plan to move on to fix reflective use in the various extensions once the core AOP change is in.

**Comment by mcculls** [comment]
FYI, the specific code in AssistedInject triggering this warning message is https://github.com/google/guice/blob/master/extensions/assistedinject/src/com/google/inject/assistedinject/FactoryProvider2.java#L900
